lib/travis/build/templates/header.sh

```diff
@@ -52,11 +53,17 @@ travis_cmd() {
   fi
 
   if [[ -n "$retry" ]]; then
-    travis_retry eval "$cmd"
+    travis_retry eval "$cmd $secure"
+    result=$?
   else
-    eval "$cmd"
+    eval "$cmd $secure"
+    result=$?
+    if [[ -n $secure && $result -ne 0 ]]; then
+      echo -e "${ANSI_RED}The previous command failed, possibly due to a malformed secure environment variable.${ANSI_CLEAR}
+${ANSI_RED}Please be sure to escape special characters such as ' ' and '$'.${ANSI_CLEAR}
+${ANSI_RED}For more information, see http://www.tldp.org/LDP/abs/html/special-chars.html.${ANSI_CLEAR}"
```

Comment:
We might want to mention our own documentation as well (or instead) https://docs.travis-ci.com/user/encryption-keys

Maybe https://docs.travis-ci.com/user/encryption-keys should point to the Bash documentation above.